### ISSUE TITLE
[ProtoFlux / UX] Prevent potentially invalid / accidental Pack In Place

### ISSUE THREAD

**Issue by Zyro1331** [Yellow-Dog-Man/Resonite-Issues#3128]
### Describe the bug?

When editing newly created ProtoFlux, you can highlight it with your laser and open your context menu. The pack in place button will appear, even if all the nodes are in the world Root.
![Image](https://github.com/user-attachments/assets/f06f71a0-4b57-4fa2-acb6-094139bf433e)

This can result in people accidentally packing their ProtoFlux into the Root slot of the world they're in. This can become a pain to try to recover from, since you should not Unpack the root slot or else it will unpack *everything* under that slot _(Including other user avatars)_. 
![Image](https://github.com/user-attachments/assets/d3744a6e-e009-4c2f-b332-aa5f37e86842)

All you can do in this situation is pick various nodes from your ProtoFlux out and unpack them individually, which can become time-consuming for large scripts.

### To Reproduce

1. Equip a ProtoFlux tool and write a simple script. 
_(It can be anything, just as long as there are enough nodes connected together to warrant packing.)_
2. Highlight a node with your laser, and open your Context Menu.
3. Select "Pack In Place".

The ProtoFlux will indeed follow your instructions and pack itself in place, even if it is still attached to the World Root.

### Expected behavior

I feel that there should be a small measure to prevent the nodes from packing under the Root slot or packing under the same Parent slot as the user's root. 
Disabling the Pack In Place button when that condition is detected would make the user experience feel a lot more solid, as I've seen plenty of people accidentally make the same mistake as me and spend several minutes unraveling it. 

Alternatively, something like issue #3127 could also help. 
As when this occurs, you could simply just press undo and not have to face the same consequences.

### Resonite Version Number

Beta 2024.10.23.15

### What Platforms does this occur on?

Windows

### What headset if any do you use?

Meta Quest 3 + SteamLinkVR

### Log Files

[MANGOVRG - 2024.10.23.15 - 2024-10-24 08_14_41.log](https://github.com/user-attachments/files/17509131/MANGOVRG.-.2024.10.23.15.-.2024-10-24.08_14_41.log)

### Reporters

@zyro1331 + The many others you can find in the Discord by searching "Pack In Place"

**Comment by epicEaston197** [comment]
small tip if you accidentally pack your protoflux to root you can unpack one node and with the protoflux tool select it that will then propagate through the entire script and unpack it

**Comment by JackTheFoxOtter** [comment]
> All you can do in this situation is pick various nodes from your ProtoFlux out and unpack them individually, which can become time-consuming for large scripts.

Actually that sounds like a bug, when you unpack an individual node it should unpack the entire node group.

**Comment by Nytra** [comment]
Yeah that might be the underlying cause of this https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/375

**Comment by epicEaston197** [comment]
> Yeah that might be the underlying cause of this [#375](https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/375)

No 
while that might be a symptom it's not the main issue the main issue is protoflux accidentally being packed to root Which shouldn't happen to begin with if there isn't a specified packing root

**Comment by JackTheFoxOtter** [comment]
I honestly don't think packing to root is too big of a deal, the issue is just that it's a little bothersome to undo if you did so accidentally. I feel like #3127 would solve the underlying problem as well.

**Comment by Zyro1331** [comment]
> I honestly don't think packing to root is too big of a deal, the issue is just that it's a little bothersome to undo if you did so accidentally. I feel like [#3127](https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/3127) would solve the underlying problem as well.

This issue is mostly for just User Experience and Quality of Life, since the core issue is mostly resolvable by doing a bit of manual work. I did mention #3127 in the issue above, as I also feel that it may work well to resolve this.

I just feel that if you are a new user learning ProtoFlux, an accidental Pack In Place can interfere and end up being confusing to untangle without external guidance. 
Plus, anyone who is doing professional work with large scripts will end up feeling like they are wasting time after this accident occurs. 
I've seen people talk about how "dangerous" Pack In Place is after experiencing this problem, which I feel is just incorrect, but I can understand where such strong feelings come from.

**Comment by shiftyscales** [comment]
I don't believe this issue to be a bug in any case- the root slot has no special treatment I'm aware of compared to any other slot with the ProtoFlux tool, and as such nothing is technically misbehaving.

For the context of accidentally packing to / unpacking the root slot- I agree #3127 and #3118 should work to handle a majority of those cases.

> All you can do in this situation is pick various nodes from your ProtoFlux out and unpack them individually, which can become time-consuming for large scripts.

I feel like it may be more beneficial to highlight this particular behaviour instead.

As others have mentioned- it is possible to highlight the single node you unpacked to unpack the rest of the chain- but I don't know how useful it is by default to unpack just a single node- it could be a good QoL addition that unpacking a single node in the inspector will unpack whatever it is connected to directly by default?

Curious for input from @Frooxius.
